packages/trackerless-network/test/unit/StreamrNode.test.ts

```diff
@@ -33,25 +27,75 @@ describe('StreamrNode', () => {
         expect(isSamePeerDescriptor(peerDescriptor, node.getPeerDescriptor()))
     })
 
-    it('can join streams', async () => {
-        await node.joinStream(streamPartId)
-        expect(node.hasStream(streamPartId)).toEqual(true)
-    })
+    describe('join and leave', () => {
 
-    it('can leave streams', async () => {
-        await node.joinStream(streamPartId)
-        expect(node.hasStream(streamPartId)).toEqual(true)
-        node.leaveStream(streamPartId)
-        expect(node.hasStream(streamPartId)).toEqual(false)
-    })
+        const streamPartId = StreamPartIDUtils.parse('stream#0')
+        const message = createStreamMessage(
+            JSON.stringify({ hello: 'WORLD' }), 
+            streamPartId, 
+            randomEthereumAddress()
+        )
 
-    it('broadcast joins stream', async () => {
-        node.broadcast(message)
-        await waitForCondition(() => node.hasStream(streamPartId))
+        beforeEach(async () => {
+            node.setStreamPartEntryPoints(streamPartId, [node.getPeerDescriptor()])
+        })
+    
+        it('can join streams', async () => {
+            await node.joinStream(streamPartId)
+            expect(node.hasStream(streamPartId)).toEqual(true)
+        })
+    
+        it('can leave streams', async () => {
+            await node.joinStream(streamPartId)
+            expect(node.hasStream(streamPartId)).toEqual(true)
+            node.leaveStream(streamPartId)
+            expect(node.hasStream(streamPartId)).toEqual(false)
+        })
+    
+        it('broadcast joins stream', async () => {
+            node.broadcast(message)
+            await waitForCondition(() => node.hasStream(streamPartId))
+        })
+    
+        it('can leave', async () => {
+            await node.joinStream(streamPartId)
+            node.leaveStream(streamPartId)
+        })
     })
 
-    it('can leave', async () => {
-        await node.joinStream(streamPartId)
-        node.leaveStream(streamPartId)
+    describe('proxied stream', () => {
```

Comment:
Add async/await for setting proxies in this test block.

Reply:
Fixed